engine/main.ml

```diff
@@ -51,6 +51,8 @@ let compare source_file target_file =
   let target = target_exec target_file in
   let repr_env =
     Source_env.build_type_repr_env source.type_decls in
+  let repr_env =
+    Source_env.ConstructorMap.add "()" (Source_env.Int 0) repr_env in
```

Comment:
I think it would be more future-proof to include `Unit` as a built-in constructor in the `constructor` type, and extend various constructor-dependent functions to also handle it. Otherwise you may need to add this logic to the type environment as well, for example (if we decide to do a pattern-matching on the unit value, and we want the `width` of this type). This would also respect the general intuition that {type,repr}_env contain information on the user-defined types.